base/applications/rapps/appview.cpp

```diff
@@ -1382,7 +1382,23 @@ BOOL CAppsListView::AddInstalledApplication(CInstalledApplicationInfo *InstAppIn
         return FALSE;
     }
 
-    HICON hIcon = (HICON)LoadIconW(hInst, MAKEINTRESOURCEW(IDI_MAIN));
+    /* Load icon from registry */
+    HICON hIcon = NULL;
+    ATL::CStringW szIconPath;
+    if (InstAppInfo->RetrieveIcon(szIconPath))
+    {
+        PathParseIconLocationW((LPWSTR)szIconPath.GetString());
```

Comment:
You'd need the return from this to get the icon index

Reply:
Depends what @oleg-dubinskiy really wants to use as the icon (either hardcodes always icon zero, or actually uses whatever icon index was specified in the icon path)

Reply:
Yes, I think hardcoded zero will be better, since according to my analysis, all apps provide executables with the main icon as first in the index (detected as `0`), so we don't need other icons here (which have `1`, `2`, `3` etc. index numbers).

Reply:
Return value is required only for the random suitable index value.

Reply:
In that case it might make sense to add a comment explaining that you deliberately ignore this?